game-core/src/main/java/org/triplea/thread/LockUtil.java

```diff
@@ -1,154 +0,0 @@
-package org.triplea.thread;
-
-import com.google.common.annotations.VisibleForTesting;
-import java.lang.ref.WeakReference;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Objects;
-import java.util.Set;
-import java.util.WeakHashMap;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.concurrent.locks.Lock;
-import lombok.extern.java.Log;
-
-/**
- * Utility class for ensuring that locks are acquired in a consistent order.
- *
- * <p>Simply use this class and call acquireLock(aLock) releaseLock(aLock) instead of lock.lock(),
- * lock.release(). If locks are acquired in an inconsistent order, an error message will be printed.
```

Comment:
My experience with this class was that it was some sort of primitive deadlock detector for cases where multiple locks were acquired on different, but concurrent, code paths.  That is, it was an attempt to enforce consistent lock ordering by preventing constructs like the following:

```
Thread A                   Thread B
--------                   --------
synchronized(lock1) {      synchronized(lock2) {
  . . .                      . . .
  synchronized(lock2) {      synchronized(lock1) {
    . . .                      . . .
  }                          }
}                          }
```

Where the lock acquisitions may happen in totally different methods, and thus not be obvious to a developer that they are introducing an inconsistent lock ordering.

I don't know if there are still such cases where the same set of locks are acquired on concurrent code paths.  If so, that would be the only reason I could see keeping this class around.

Reply:
I agree but looking at TileManager and Tile I believe the synchronised blocks are pretty safe: Tile operations are only called from newly created threads AFAIK, so the lock order is consistent there, at least for the Tile class, TileManager is similar where the operations are encapsulated well from my point of view.
The only thing I'd agree where this class might help is for the whole GameData thing. However it only really works if all places are using LockUtil.

If we want to be absolutely sure we could also implement a background thread that regularly runs a real deadlock check every couple of minutes, which would result in less code and be a more sophisticated approach to deadlock detection.

Reply:
> The only thing I'd agree where this class might help is for the whole GameData thing. However it only really works if all places are using LockUtil.

Agreed.  I suspect, at some point, the intent of using `LockUtil` consistently was simply neglected.

> If we want to be absolutely sure we could also implement a background thread that regularly runs a real deadlock check every couple of minutes, which would result in less code and be a more sophisticated approach to deadlock detection.

My guess is `LockUtil` was created in response to a scenario whereby inconsistent lock ordering was being introduced by a developer, but it didn't result in a deadlock on the dev's machine due to choice timing.  Then, upon deployment to a user's machine, the timing changed such that the deadlock potential was realized, and that caused all sorts of headaches for the devs to debug.  This class then would have been introduced as an attempt to detect a _potential_ deadlock rather than simply a tool to debug a _realized_ deadlock (which modern JVM toolsets now provide anyway).

Reply:
Haven't thought of that, good consideration